examples/webgl_impostors.html

```diff
@@ -0,0 +1,227 @@
+				const lights = [ hemLight, directionalLight ];
+
+				// ground plane
+
+				const groundGeometry = new THREE.PlaneGeometry( sceneWidth * 2, sceneWidth * 2 );
+				const groundMaterial = new THREE.MeshStandardMaterial( { color: 'grey' } );
+				const groundMesh = new THREE.Mesh( groundGeometry, groundMaterial );
+				groundMesh.rotation.x = - Math.PI / 2;
+				scene.add( groundMesh );
+
+				// cars and impostors setup
+
+				const dracoLoader = new DRACOLoader().setDecoderPath( 'js/libs/draco/gltf/' );
+				const loader = new GLTFLoader().setDRACOLoader( dracoLoader );
+
+				loader.load( './models/gltf/ferrari.glb', (gltf) => {
+
+					for ( let i = 0 ; i < carsNumber ; i ++ ) {
+
+						const car = gltf.scene.clone();
+
+						car.position.set(
+							Math.random() * sceneWidth * 2 - sceneWidth,
+							0,
+							Math.random() * sceneWidth * 2 - sceneWidth
+						);
+
+						car.rotation.y = -Math.PI / 2;
+						car.direction = Math.random() * 0.1 + 0.15;
+
+						scene.add( car );
+						cars.push( car );
+
+						/* IMPOSTOR CODE */
+
+						const impostor = new Impostor( car, camera, renderer, scene );
+
+						impostors.push( impostor );
+
+						// impostor options
+
+						impostor.lights = lights;
+
+						impostor.impostureDistance = 90;
+
+						impostor.setSize( 128 );
+
+						impostor.maxAngle = 0.3;
+
+					}
+
+				} );
+
+				// interactive GUI
+
+				const normalMaterial = new THREE.MeshNormalMaterial();
+
+				const params = {
+					enable: true,
+					show: false,
+					dist: 90
+				};
+
+				const gui = new GUI();
+
+				gui.add( params, 'enable' ).name( 'enable impostors' ).onChange( ( bool ) => {
+
+					if ( bool ) {
+
+						impostors.forEach( impostor => impostor.enabled = true );
+
+					} else {
+
+						impostors.forEach( impostor => impostor.enabled = false );
+
+					}
+
+				} );
+
+				gui.add( params, 'show' ).name( 'show impostors' ).onChange( ( bool ) => {
+
+					if ( bool ) {
+
+						impostors.forEach( impostor => {
+
+							impostor.userData.oldMaterial = impostor.material;
+							impostor.material = normalMaterial;
+
+						} );
+
+					} else {
+
+						impostors.forEach( impostor => {
+
+							impostor.material = impostor.userData.oldMaterial;
+
+						} );
+
+					}
+
+				} );
+
+				gui.add( params, 'dist', 20, 500 ).step( 1 ).name( 'min distance' ).onChange( ( val ) => {
+
+					impostors.forEach( impostor => {
+
+						impostor.impostureDistance = val;
+
+					} );
+
+				} );
+
+			}
+
+			function onWindowResize() {
+
+				camera.aspect = window.innerWidth / window.innerHeight;
+				camera.updateProjectionMatrix();
+
+				renderer.setSize( window.innerWidth, window.innerHeight );
+
+			}
+
+			function animate() {
+
+				requestAnimationFrame( animate, renderer.domElement );
+
+				cars.forEach( (car) => {
+
+					car.position.x += car.direction;
+
+					if ( car.position.x > sceneWidth ) {
+
+						car.position.x = - sceneWidth;
+
+					}
+
+				} );
+
+				Impostor.updateAll();
```

Comment:
How about using `onBeforeRender()` for the update logic?

Reply:
I agree it would be nice to get rid of `Impostor.updateAll`.
I just tried `onBeforeRender()`, but it seems it's not called when the mesh doesn't get rendered, so it's not called while the object to impersonate is visible and the impostor is invisible. Is there another callback I could use that get called on not rendered objects ? Or an event that get fired, maybe by the renderer ?

Reply:
Or maybe I should keep the impostor `visible == true` even when it's not needed, but swap its material or geometry so it would not be drawn.
It adds a draw call for nothing when it's close to the camera though, so I don't know...

Reply:
>  Is there another callback I could use that get called on not rendered objects ?

Thought about this for a while and maybe introducing `Object3D.onProject()` could be a solution. Right now `THREE.LOD` does this in `projectObject()`:

https://github.com/mrdoob/three.js/blob/ec5fb63a0b557474844f458065e5b26a22667e4e/src/renderers/WebGLRenderer.js#L1121-L1125

This line could be removed if `THREE.LOD` implements `onProject()`. `THREE.Impostor` could use it, too.

Reply:
> `THREE.Impostor` could use the same approach, too, right?

Yes `THREE.Impostor` could definitely use this to ditch `Impostor.updateAll`.

Reply:
@Mugen87 I can submit the PR for `Object3D.onProject()` if you want

Reply:
Let's wait what @mrdoob thinks about it.